### PR TITLE
RT: Split disruption from chaos

### PR DESCRIPTION
**Pull request by antoine-de**
technical refactoring before some more realtime handling.
- split fill_diruption_from_chaos to separate the navitia::Disruption creation and the change of the model
- rename ::new_disruption to ::disruption
- move some code

http://jira.canaltp.fr/browse/NAVP-357

### CODE REVIEW

**Comment by xlqian** [comment]
<sup>**[source/disruption/traffic_reports_api.cpp, line 112 [r1]](https://reviewable.io:443/reviews/canaltp/navitia/1228#-K19lTqTjBsKdECp6HSK)** ([raw file](https://github.com/canaltp/navitia/blob/e5ad1ba1860d780083bff23c3991e62fd354e9f9/source/disruption/traffic_reports_api.cpp#L112)):</sup>
`std::move` or `emplace_back`?



---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/canaltp/navitia/1228)

File: source/disruption/traffic_reports_api.cpp [hunk 1]
```diff
@@ -195,12 +247,70 @@ void Disruption::disruptions_list(const std::string& filter,
...
+        for (const auto& line_item: dist.lines) {
+            pbnavitia::Line* pb_line = pb_disruption->add_lines();
+            navitia::fill_pb_object(line_item.first, d, pb_line, depth-1, bt::not_a_date_time, action_period, false);
+            for(const auto& impact: line_item.second){
+                fill_message(*impact, d, pb_line, depth-1, now_dt, action_period);
+            }
+        }
+        for (const auto& sa_item: dist.stop_areas) {
+            pbnavitia::StopArea* pb_stop_area = pb_disruption->add_stop_areas();
+            navitia::fill_pb_object(sa_item.first, d, pb_stop_area, depth-1, bt::not_a_date_time, action_period, false);
```
Comment (pbougue) [inline source/disruption/traffic_reports_api.cpp]: line length

File: source/disruption/traffic_reports_api.cpp [hunk 2]
```diff
@@ -195,12 +247,70 @@ void Disruption::disruptions_list(const std::string& filter,
...
+    if (pb_response.disruptions_size() == 0) {
+        fill_pb_error(pbnavitia::Error::no_solution, "no solution found for this disruption",
+        pb_response.mutable_error());
+        pb_response.set_response_type(pbnavitia::NO_SOLUTION);
+    }
+    return pb_response;
 }
 
-}}
+}}//namespace
```
Comment (pbougue) [inline source/disruption/traffic_reports_api.cpp]: minor: `navitia::disruption`

File: source/disruption/traffic_reports_api.cpp [hunk 3]
```diff
@@ -28,13 +28,66 @@ IRC #navitia on freenode
...
+    const type::Network* network = nullptr;
+    DisruptionSet network_disruptions;
+    //we use a vector of pair because we need to sort by the priority of the impacts
+    std::vector<std::pair<const type::Line*, DisruptionSet>> lines;
+    std::vector<std::pair<const type::StopArea*, DisruptionSet>> stop_areas;
+};
+
+class TrafficReport {
+private:
+
```
Comment (pbougue) [inline source/disruption/traffic_reports_api.cpp]: --line

File: source/disruption/traffic_reports_api.cpp [hunk 4]
```diff
@@ -28,13 +28,66 @@ IRC #navitia on freenode
...
+    void add_networks(const std::vector<type::idx_t>& network_idx,
+                      const type::Data &d,
+                      const boost::posix_time::ptime now);
+    void add_lines(const std::string& filter,
+                      const std::vector<std::string>& forbidden_uris,
+                      const type::Data &d,
+                      const boost::posix_time::ptime now);
+    void sort_disruptions();
+public:
```
Comment (pbougue) [inline source/disruption/traffic_reports_api.cpp:73]: ++line

File: source/type/type.h [hunk 5]
```diff
@@ -302,16 +302,16 @@ struct HasMessages{
...
 
-    std::vector<boost::weak_ptr<new_disruption::Impact>> get_impacts() const {
+    std::vector<boost::weak_ptr<disruption::Impact>> get_impacts() const {
         return impacts;
     }
 
-    void remove_impact(const boost::shared_ptr<new_disruption::Impact>& impact) {
-        auto it = std::find_if(impacts.begin(), impacts.end(),[&impact](const boost::weak_ptr<new_disruption::Impact>& i) {
+    void remove_impact(const boost::shared_ptr<disruption::Impact>& impact) {
+        auto it = std::find_if(impacts.begin(), impacts.end(),[&impact](const boost::weak_ptr<disruption::Impact>& i) {
```
Comment (pbougue) [inline source/type/type.h]: line length

File: source/disruption/traffic_reports_api.cpp (hunk 2 above)
Comment (antoine-de) [inline source/disruption/traffic_reports_api.cpp]: not convinced

File: source/disruption/traffic_reports_api.cpp (hunk 4 above)
Comment (antoine-de) [inline source/disruption/traffic_reports_api.cpp:73]: not convinced

File: source/disruption/traffic_reports_api.cpp (hunk 2 above)
Comment (TeXitoi) [inline source/disruption/traffic_reports_api.cpp]: I prefere with the name of the namespace, but it's minor.

File: source/disruption/traffic_reports_api.cpp (hunk 4 above)
Comment (pbougue) [inline source/disruption/traffic_reports_api.cpp:73]: minor, and it's above that I think it's missing. as you wish

**Comment by xlqian** [comment]
<sup>**[source/kraken/apply_disruption.h, line 45 [r2]](https://reviewable.io:443/reviews/canaltp/navitia/1228#-K1Df4v9aJ0-8Ue_y5Dx)** ([raw file](https://github.com/canaltp/navitia/blob/a638551ac7d2d28d9785ed99e06c4ea49dc33064/source/kraken/apply_disruption.h#L45)):</sup>
if the pointer cannot be null, i'd prefer passing by ref, i don't know if it's possible...



---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/canaltp/navitia/1228)

**Comment by xlqian** [comment]
Reviewed 31 of 33 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks pending.

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/canaltp/navitia/1228)

File: source/kraken/apply_disruption.cpp [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+        apply_impacts_visitor(impact, pt_data, meta, "add") {}
+
+    ~add_impacts_visitor() {}
+    add_impacts_visitor(const add_impacts_visitor&) = default;
+
+    using apply_impacts_visitor::operator();
+
+    bool func_on_vj(nt::VehicleJourney& vj) {
+        nt::ValidityPattern tmp_vp(*vj.adapted_validity_pattern());
+        for (auto period : impact->application_periods) {
```
Comment (pbougue) [inline source/kraken/apply_disruption.cpp:132]: const &?

File: source/kraken/apply_disruption.cpp [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+    // other disruptions after
+    bool func_on_vj(nt::VehicleJourney& vj) {
+        vj.validity_patterns[nt::RTLevel::Adapted] = vj.validity_patterns[nt::RTLevel::Base];
+        ++ nb_vj_reassigned;
+        const auto& impact = this->impact;
+        boost::range::remove_erase_if(vj.impacted_by,
+            [&impact](const boost::weak_ptr<nt::disruption::Impact>& i) {
+                auto spt = i.lock();
+                return (spt) ? spt == impact : true;
+        });
```
Comment (pbougue) [inline source/kraken/apply_disruption.cpp:201]: indent

**Comment by antoine-de** [comment]
Jenkins, test this please

**Pull request merged**